docs/apis/plugintypes/repository/index.mdx

```diff
@@ -0,0 +1,975 @@
+
+As a plug-in developer, if you set dynload flag as **true**, you should return files and folders (set children as a null array) in current path only instead of building the whole file tree.
+
+The use of the **object** tag, instead of returning a **list** of files, allows you to embed an external file chooser within the repository panel. See [[Repository plugins embedding external file chooser]] for details about how to do this.
+
+### User login (optional)
+
+If the plugin requires login from the user at the time when they use it, then these functions can be used.
+
+#### print_login
+
+Returns an array of the elements required in the login form. If no login form is required, then the default implementation of this will redirect to the files list. If $this->options[is not set, then an HTML-snippet with the login fields (but not the form tags) should be output, instead of returning the form details.
+
+<details>
+  <summary>View example</summary>
+  <div>
+
+```php
+public function print_login() { // From repository_pluginname
+    global $OUTPUT;
+
+    if ($this->options['ajax']('ajax'])) {
+        $user_field = (object) [
+            'label' => get_string('username', 'repository_pluginname') . ': ',
+            'id'    => 'pluginname_username',
+            'type'  => 'text',
+            'name'  => 'al_username',
+        ];
+
+        $passwd_field = (object) [
+            'label' => get_string('password', 'repository_pluginname') . ': ',
+            'id'    => 'pluginname_password',
+            'type'  => 'password',
+            'name'  => 'al_password',
+        ];
+
+        $ret = [];
+        $ret['login'] = [$user_field, $passwd_field];
+        return $ret;
+    } else { // Non-AJAX login form - directly output the form elements.
+        // Print the login form HTML including the input username and password fields.
+        $loginform = new repository_pluginname\output\login();
+        echo $OUTPUT->render($loginform);
+        // Example of a login form:
+        // <label> Username </label>
+        // <input type="text" name="al_username" />
+        // <label> Password </label>
+        // <input type="password" name="al_password" />
+        // <input type="submit" value="Enter" />
+    }
+}
+```
+
+  </div>
+</details>
+
+This will help to generate a form by file picker which contains user name and password input elements.
+
+If your login form is static and never changes, you can add __$ret\['allowcaching'\]('login']) = true;__ and filepicker will not send the request to the server every time user opens the login/search form.
+
+For plugins that do not fully process the login via a popup window, the submitted details can be retrieved, from within the '\__construct' function, via $submitted = optional_param('fieldname', [PARAM_INT/PARAM_TEXT).
+
+<details>
+  <summary>View example</summary>
+  <div>
+
+```php title="Taken from repository_alfresco"
```

Comment:
Use the filename instead of this title.
Also, if possible can we use a repo which is in core?

```suggestion
```php title="lib/alfresco/lib.php"
```